internal/provider/incident_catalog_type_data_source.go

```diff
@@ -0,0 +1,125 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/incident-io/terraform-provider-incident/internal/apischema"
+	"github.com/incident-io/terraform-provider-incident/internal/client"
+	"github.com/samber/lo"
+)
+
+var (
+	_ datasource.DataSource              = &IncidentCatalogTypeDataSource{}
+	_ datasource.DataSourceWithConfigure = &IncidentCatalogTypeDataSource{}
+)
+
+func NewIncidentCatalogTypeDataSource() datasource.DataSource {
+	return &IncidentCatalogTypeDataSource{}
+}
+
+type IncidentCatalogTypeDataSource struct {
+	client *client.ClientWithResponses
+}
+
+func (i *IncidentCatalogTypeDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This data source provides information about a catalog type.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: apischema.Docstring("CatalogTypeV2ResponseBody", "id"),
+				Computed:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: apischema.Docstring("CatalogV2CreateTypeRequestBody", "name"),
+				Optional:            true,
+				Computed:            true,
+			},
+			"type_name": schema.StringAttribute{
+				MarkdownDescription: apischema.Docstring("CatalogV2CreateTypeRequestBody", "type_name"),
+				Optional:            true,
+				Computed:            true,
+			},
+			"description": schema.StringAttribute{
+				MarkdownDescription: apischema.Docstring("CatalogTypeV2ResponseBody", "description"),
+				Computed:            true,
+			},
+			"source_repo_url": schema.StringAttribute{
```

Comment:
Is this needed? I know the API returns it at the moment, but we introduced managed resources in the backend a week or so ago, and if it's not needed by this change I'd avoid including it in case it causes us migration trouble in the future.